tutorials/counter/README.md

```diff
@@ -0,0 +1,227 @@
+      - [Deploying the contract](#4-deploying-your-sway-contract)
+    - [Using the contract in a React project](#5-using-your-sway-contract-in-a-react-project)
+- [Contributing](#contributing)
+- [License](#license)
+
+## About
+
+In this tutorial, we will build a counter application, using storage on our Sway contract to keep track of a stored number that can be retrieved/ modified as part of our React app. This demonstrates a Sway Program called Smart Contract, and will show you the basics of using Sway with the [TypeScript SDK](https://fuellabs.github.io/fuels-ts/).
+
+<p align="center">
+  ![Demo Screen Shot](./img/preview.png)
+</p>
+
+<p align="right">(<a href="#top">back to top</a>)</p>
+
+## Getting Started
+
+We recommend that you first read through and understand the basics of [Sway](https://fuellabs.github.io/sway/latest/), the DSL for the Fuel Virtual Machine.
+
+### Prerequisites
+
+Your machine will need to have a few things installed in order to run a local fuel node and this React application:
+
+- [The latest LTS version of Node.js](https://nodejs.org/)
+- [The latest forc and fuel-core binaries](https://fuellabs.github.io/sway/latest/introduction/installation.html#dependencies)
+
+### Installation
+
+Open this demo application
+
+```sh
+cd tutorials/counter
+```
+
+Install NPM packages
+
+```sh
+npm i
+```
+
+### Usage
+
+To quickly run this demo application, follow these steps, or see detailed tutorial below
+
+Run React application in dev mode, make sure to configure a new [`.env`](./.env.example) file.
+
+```sh
+npm run dev
+```
+
+View in local browser
+
+```sh
+open http://localhost:3000
+```
+
+### Walkthrough
+
+This application tutorial consists of five main milestones on your journey to writing your first Sway + React app:
+
+1. Creating a deployable Sway program - in this example we will create a [Smart Contract](https://fuellabs.github.io/sway/latest/sway-program-types/smart_contracts.html)
+2. Compiling your Sway contract
+3. Generating TypeScript for the `counter` program
+4. Deploying the `counter` program
+5. Using your Sway contract in a React project
+
+#### 1. Creating a deployable Sway program
+
+The counter program is based off the example in the [Sway book](https://fuellabs.github.io/sway/latest/examples/counter.html).
+
+We require 3 functions for our application.
+
+1. Incrementing the counter
+2. Decrementing the counter
+3. Getting the current value of the counter
+
+Take a look at the following [abi.sw](src/abi.sw) of the program:
+
+```rust
+    fn increment();
+    fn decrement();
+    fn get_counter() -> u64;
+```
+
+#### 2. Compiling your Sway contract
+
+The Sway program source can be compiled into binary and generate an Application Binary Interface (ABI) JSON file that allows typed and strict interoperability between Fuel and development languages like Rust and TypeScript.
+
+While inside this project directory, use this command to take the contents of [main.sw](src/main.sw) and compile it.
```

Comment:
Could you be a bit more specific about "this project directory"? Someone might not know where they should be running the command from. It's better to explicitly state the path